apps/hash-frontend/src/pages/entities.page.tsx

```diff
@@ -157,30 +224,41 @@ const EntitiesPage: NextPageWithLayout = () => {
                 loading={loading}
               />
             </Tabs>
-            {isViewAllPagesPage && hashInstance?.properties.pagesAreEnabled ? (
-              <Button
-                endIcon={<FileCirclePlusRegularIcon />}
-                variant="tertiary_quiet"
-                onClick={createPage}
-                sx={{
-                  color: ({ palette }) => palette.gray[90],
-                  fontSize: 14,
-                  padding: 0,
-                  ":hover": {
-                    background: "transparent",
-                    color: ({ palette }) => palette.blue[70],
-                    [`.${buttonClasses.endIcon}`]: {
-                      color: ({ palette }) => palette.blue[70],
-                    },
-                  },
-                  [`.${buttonClasses.endIcon}`]: {
-                    color: ({ palette }) => palette.blue[70],
-                  },
-                }}
-              >
-                Create a page
-              </Button>
-            ) : null}
+            <Fade in={displayCreateEntityButton}>
+              {isViewAllPagesPage ? (
```

Comment:
If someone visits this page filtered to `Document` or `Canvas` they will get a create button which takes them to the regular entity editor – I think we might be better off making `CreateButtons` into a proper component which then renders different stuff depending on the entityTypeId filtered to, e.g.

1. if `Document`, show document button
2. If `Canvas`, show canvas button
3. if `Page`, show both
4. show default, with special copy for files

We can then hide all the special casing inside the `CreateButton` component (including checking if pages are enabled), since it isn't needed outside of there

Reply:
That's a fair point, I've addressed this in https://github.com/hashintel/hash/pull/3923/commits/32e86e7d9a0250b2253daa78fe86084cdb9ba323. I also spotted that the `Canvas` page was being pluralized weirdly to `Canvass`, so added the `pluralize` package in https://github.com/hashintel/hash/pull/3923/commits/f567d5bc21dfc5388dddaea18ae0126ae690880c